base/workflow/R/run.write.configs.R

```diff
@@ -26,15 +26,22 @@ run.write.configs <- function(settings, write = TRUE, ens.sample.method = "unifo
                               posterior.files = rep(NA, length(settings$pfts)), 
                               overwrite = TRUE) {
   
-  con <- PEcAn.DB::db.open(settings$database$bety)
-  on.exit(PEcAn.DB::db.close(con))
+
   
   ## Which posterior to use?
   for (i in seq_along(settings$pfts)) {
     ## if posterior.files is specified us that
     if (is.na(posterior.files[i])) {
       ## otherwise, check to see if posteriorid exists
       if (!is.null(settings$pfts[[i]]$posteriorid)) {
+        
+        tryCatch({
+          con <- PEcAn.DB::db.open(...)
+          on.exit(PEcAn.DB::db.close(con)
```

Comment:
Missing close paren here (thanks, Travis!). May as well add the `add = TRUE` as well while you're at it.

```suggestion
          on.exit(PEcAn.DB::db.close(con), add = TRUE)
```

https://travis-ci.org/PecanProject/pecan/jobs/590413421#L3177